contracts/optimistic-ethereum/libraries/codec/Lib_EIP155Tx.sol

```diff
@@ -0,0 +1,119 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >0.5.0 <0.9.0;
+pragma experimental ABIEncoderV2;
+
+/* Library Imports */
+import { Lib_RLPReader } from "../rlp/Lib_RLPReader.sol";
+import { Lib_RLPWriter } from "../rlp/Lib_RLPWriter.sol";
+
+import { console } from "hardhat/console.sol";
+
+library Lib_EIP155Tx {
+    struct EIP155Tx {
+        uint256 nonce;
+        uint256 gasPrice;
+        uint256 gasLimit;
+        address to;
+        uint256 value;
+        bytes data;
+        uint8 recoveryParam;
```

Comment:
This is basically just caching the recovery param so that it is only computed once instead of needing to compute it when calling `sender` which is only called once and is always called. Might save gas to use it depending on how well packed this struct can be made

Reply:
Right. Given a chain ID, the recovery param will never change. So it seems a bit easier to compute this once (we *do* actually end up computing this in two different places otherwise).